bundles/org.openhab.binding.lgwebos/src/main/java/org/openhab/binding/lgwebos/internal/handler/LGWebOSTVSocket.java

```diff
@@ -365,7 +370,25 @@ public void onMessage(String message) {
                     break;
                 }
                 try {
-                    request.processResponse(response.getPayload().getAsJsonObject());
+                    JsonObject jsonResponse = response.getPayload().getAsJsonObject();
```

Comment:
Instead of handling this in onMessage. Let's do it this way to keep those concerns in one coherent in one place:

```
 public void powerOff(final ResponseListener<CommandConfirmation> listener) {
        String uri = "ssap://system/turnOff";

        ResponseListener<CommandConfirmation> interceptor = new ResponseListener<CommandConfirmation>() {

            @Override
            public void onSuccess(CommandConfirmation confirmation) {
                if (confirmation.getReturnValue() && state == State.REGISTERED) {
                    setState(State.DISCONNECTING);
                }
                listener.onSuccess(confirmation);
            }

            @Override
            public void onError(String message) {
                listener.onError(message);
            }
        };
        ServiceCommand<CommandConfirmation> request = new ServiceCommand<>(uri, null,
                x -> GSON.fromJson(x, CommandConfirmation.class), interceptor);
        sendCommand(request);
    }
```

same for the application listener

Reply:
Yes, that looks cleaner like this.